src/components/10-atoms/input-text/index.js

```diff
@@ -190,6 +189,22 @@ class AXAInputText extends AXAPopupMixin(NoShadowDOM) {
     if (this.maxLength) {
       this.modelCounter = this.getCounterText;
     }
+
+    // means that a automatic prefelling function did fill it. Re-evaluate the component
+    // if a user types the chars, the charsLeft always decrease incrementally. With
+    // autocomplete instead is decreased all at once, therefore truncate it
+    if (this.charsLeft < 0 && this.maxLength) {
+      const { nativeInput } = this;
+      const value = nativeInput.value.substring(0, this.maxLength - 1);
+
+      // set value of native input element
+      nativeInput.value = value;
+      // set model value
+      this.modelValue = value;
+      // requestr update with the new value
+      this.requestUpdate('value', value);
+      this.modelCounter = this.getCounterText;
```

Comment:
Could you move code from line 189-191 to line 209? If you could, this line 206 is not necessary.